tests/unit/py_proxy/views_test.py

```diff
@@ -105,86 +114,176 @@ def test_pdf_renders_parameters_in_pdf_template(
     @pytest.mark.parametrize(
-        "content_type,redirect_location",
+        "content_type,redirect_url",
         [
-            ("application/pdf", "/pdf/http://thirdparty.url"),
-            ("application/x-pdf", "/pdf/http://thirdparty.url"),
-            ("text/html", "http://via.hypothes.is/http://thirdparty.url"),
+            ("application/pdf", "http://localhost/pdf/https://thirdparty.url"),
+            ("application/x-pdf", "http://localhost/pdf/https://thirdparty.url"),
+            ("text/html", "http://via.hypothes.is/https://thirdparty.url"),
         ],
     )
-    @httpretty.activate
     def test_redirects_based_on_content_type_header(
-        self, pyramid_request, content_type, redirect_location
+        self, pyramid_config, content_type, redirect_url
     ):
-        httpretty.register_uri(
-            httpretty.GET,
-            "http://thirdparty.url",
-            body="{}",
-            adding_headers={"Content-Type": content_type},
+        request = self._create_request(
+            request_url="/https://thirdparty.url",
+            thirdparty_url="https://thirdparty.url",
+            content_type=content_type,
+            pyramid_config=pyramid_config,
         )
 
-        result = views.content_type(pyramid_request)
+        result = views.content_type(request)
+
+        assert result.location == redirect_url
 
-        assert result.location == redirect_location
+    @pytest.mark.parametrize(
+        "query_param", ["via.request_config_from_frame", "via.open_sidebar"]
+    )
+    @pytest.mark.xfail(raises=AssertionError)
+    def test_does_not_pass_via_query_params_to_thirdparty_server(
+        self, pyramid_config, query_param
+    ):
+        request = self._create_request(
+            request_url="/https://thirdparty.url?"
+            "via.request_config_from_frame=lms.hypothes.is&via.open_sidebar=1",
+            thirdparty_url="https://thirdparty.url",
+            content_type="application/pdf",
+            pyramid_config=pyramid_config,
+        )
 
-    @pytest.fixture
-    def pyramid_request(self, pyramid_request):
-        pyramid_request.matchdict = {"url": "http://thirdparty.url"}
+        views.content_type(request)
 
-        def route_url(path, pdf_url):
-            return f"/{path}/{pdf_url}"
+        # pylint: disable-msg=E1101
+        assert query_param not in httpretty.last_request().path
 
-        pyramid_request.route_url = route_url
-        return pyramid_request
+    # pylint: disable-msg=too-many-arguments
+    def _create_request(
+        self, request_url, thirdparty_url, content_type, pyramid_config,
+    ):
+        httpretty.register_uri(
+            httpretty.GET,
+            thirdparty_url,
+            body="{}",
+            adding_headers={"Content-Type": content_type},
+        )
+        request = Request.blank(request_url)
+        request.registry = pyramid_config.registry
+        request.matchdict = {"url": thirdparty_url}
+        return request
```

Comment:
Optional, but you could make this helper function easier to call and remove some test code duplication by using fixtures.

This has a few advantages:

* The test methods no longer need to pass `pyramid_config` to `make_pyramid_request()` (one less argument to pass each time you call it)
* The test methods no longer need to take `pyramid_config` as a test method argument
* Code duplication around creating request objects (setting `pyramid_request.registry = pyramid_config.registry`) is removed

The way you do it is:

1. Replace the `TestContentTypeRoute._create_request()` helper method with a `TestContentTypeRoute.make_pyramid_request` fixture (with `@pytest.fixture`)
 This allows `make_pyramid_request()` to have a shorter call signature than `self._create_request()` (no `self` and no `pyramid_config` argument)

2. Make the `TestContentTypeRoute.make_pyramid_request` depend on and override the module-level `make_pyramid_request` fixture that I suggested in my other comment above. This removes some test code duplication.

It would look something like this:

```python
class TestContentTypeRoute:
    def test_redirects_based_on_content_type_header(self, make_pyramid_request, content_type, redirect_url):
        pyramid_request = make_pyramid_request(
            request_url="/https://thirdparty.url",
            thirdparty_url="https://thirdparty.url",
            content_type=content_type,
        )
        ...

    ...

    @pytest.fixture
    def make_pyramid_request(self, make_pyramid_request):
        def _make_pyramid_request(request_url, thirdparty_url, content_type):
            httpretty.register_uri(
                httpretty.GET,
                thirdparty_url,
                body="{}",
                adding_headers={"Content-Type": content_type},
            )
            pyramid_request = make_pyramid_request(request_url)
            pyramid_request.matchdict = {"url": thirdparty_url}
            return pyramid_request

        return _make_pyramid_request

@pytest.fixture
def make_pyramid_request(pyramid_config):
    def _make_pyramid_request(path):
        pyramid_request = Request.blank(path)
        pyramid_request.registry = pyramid_config.registry
        return pyramid_request
    return _make_pyramid_request
```

Reply:
Ah ok that makes sense. I was trying to remember if there was a way to do this in a fixture that accepted args but now I'm remembering...return a func.